hashing/ripemd_160.cpp

```diff
@@ -0,0 +1,434 @@
+     * @param data string data to be hashed
+     * @return hex of hash in string format
+     */
+    std::string hash(std::string data) {
+        // convert string to 4-byte words including padding
+        std::vector<uint32_t> word_data = string_to_word_data(data);
+
+        // holds the current 512 bits of data to be processed
+        std::vector<uint32_t> current_block_data;
+
+        // initial state
+        int64_t h_0 = 0x67452301;
+        int64_t h_1 = 0xefcdab89;
+        int64_t h_2 = 0x98badcfe;
+        int64_t h_3 = 0x10325476;
+        int64_t h_4 = 0xc3d2e1f0;
+
+        int32_t A = 0, B = 0, C = 0, D = 0, E = 0, T = 0;
+        int32_t A_dash = 0, B_dash = 0, C_dash = 0, D_dash = 0, E_dash = 0,
+                T_dash = 0;
+
+        // for each 512 bits of data
+        for (uint64_t i = 0; i < word_data.size(); i += 16) {
+            current_block_data = std::vector<uint32_t>(
+                word_data.begin() + i, word_data.begin() + i + 16);
+
+            A = h_0;
+            B = h_1;
+            C = h_2;
+            D = h_3;
+            E = h_4;
+
+            A_dash = h_0;
+            B_dash = h_1;
+            C_dash = h_2;
+            D_dash = h_3;
+            E_dash = h_4;
+
+            // parallely process the 80 rounds
+            std::thread compute_for_block_thread(&compute_for_block, this, &A,
+                                                 &B, &C, &D, &E, &T,
+                                                 std::ref(current_block_data));
+            std::thread compute_for_block_dash_thread(
+                &compute_for_block_dash, this, &A_dash, &B_dash, &C_dash,
+                &D_dash, &E_dash, &T_dash, std::ref(current_block_data));
+
+            compute_for_block_thread.join();
+            compute_for_block_dash_thread.join();
+
+            // update the current state after 80 rounds
+            T = h_1 + C + D_dash;
+            h_1 = h_2 + D + E_dash;
+            h_2 = h_3 + E + A_dash;
+            h_3 = h_4 + A + B_dash;
+            h_4 = h_0 + B + C_dash;
+            h_0 = T;
+        }
+
+        // convert the bytes in little endian format to hex in string format
+        std::string hash = "";
+        hash = hash + uint32_t_to_string(h_0);
+        hash = hash + uint32_t_to_string(h_1);
+        hash = hash + uint32_t_to_string(h_2);
+        hash = hash + uint32_t_to_string(h_3);
+        hash = hash + uint32_t_to_string(h_4);
+
+        return hash;
+    }
+};
+}  // namespace hashing
+
+/**
+ * @brief self test implementation
+ * @return void
+ */
+static void check_me(std::string data, std::string expected_hash) {
+    hashing::RIPEMD160 obj;
+
+    std::cout << "Hashing the data " + data + "\n";
+    std::cout << "Obtained hash : " + obj.hash(data) + "\n";
+    std::cout << "Expected hash : " + expected_hash + "\n";
+    assert(obj.hash(data).compare(expected_hash) == 0);
+    std::cout
+        << "**********************************************************\n\n";
```

Comment:
Output is unnecessary probably could convert it into an inline expression

Reply:
I have some warnings in functions with switch cases that the control may reach the end of a fucntion without returning anything. But all of them are internal functions and it can be proved that the control can never reach the end of the function without returning anything.  How can we handle this?

can this be ignored or should I handle to not produce any warnings?

Example of the warning is provided below. 

C-Plus-Plus\hashing\ripemd_160.cpp:66:5: warning: control reaches end of non-void function [-Wreturn-type]
   66 |     }
      |     ^